conans/client/conan_api_v2.py

```diff
@@ -0,0 +1,1358 @@
+from conans.client.userio import UserIO
+from conans.errors import (ConanException, RecipeNotFoundException,
+                           PackageNotFoundException, NoRestV2Available, NotFoundException)
+from conans.model.editable_layout import get_editable_abs_path
+from conans.model.graph_info import GraphInfo, GRAPH_INFO_FILE
+from conans.model.graph_lock import GraphLockFile, LOCKFILE
+from conans.model.ref import ConanFileReference, PackageReference, check_valid_ref
+from conans.model.version import Version
+from conans.model.workspace import Workspace
+from conans.paths import BUILD_INFO, CONANINFO, get_conan_user_home
+from conans.paths.package_layouts.package_cache_layout import PackageCacheLayout
+from conans.search.search import search_recipes
+from conans.tools import set_global_instances
+from conans.unicode import get_cwd
+from conans.util.conan_v2_mode import CONAN_V2_MODE_ENVVAR
+from conans.util.env_reader import get_env
+from conans.util.files import exception_message_safe, mkdir, save_files
+from conans.util.log import configure_logger
+from conans.util.tracer import log_command, log_exception
+
+default_manifest_folder = '.conan_manifests'
+
+
+class ProfileData(namedtuple("ProfileData", ["profiles", "settings", "options", "env"])):
+    def __bool__(self):
+        return bool(self.profiles or self.settings or self.options or self.env)
+    __nonzero__ = __bool__
+
+
+def api_method(f):
+    def wrapper(api, *args, **kwargs):
+        quiet = kwargs.pop("quiet", False)
+        try:  # getcwd can fail if Conan runs on an unexisting folder
+            old_curdir = os.getcwd()
+        except EnvironmentError:
+            old_curdir = None
+        old_output = api.user_io.out
+        quiet_output = ConanOutput(StringIO(), color=api.color) if quiet else None
+        try:
+            api.create_app(quiet_output=quiet_output)
+            log_command(f.__name__, kwargs)
+            with environment_append(api.app.cache.config.env_vars):
+                return f(api, *args, **kwargs)
+        except Exception as exc:
+            if quiet_output:
+                old_output.write(quiet_output._stream.getvalue())
+                old_output.flush()
+            msg = exception_message_safe(exc)
+            try:
+                log_exception(exc, msg)
+            except BaseException:
+                pass
+            raise
+        finally:
+            if old_curdir:
+                os.chdir(old_curdir)
+    return wrapper
+
+
+def _make_abs_path(path, cwd=None, default=None):
+    """convert 'path' to absolute if necessary (could be already absolute)
+    if not defined (empty, or None), will return 'default' one or 'cwd'
+    """
+    cwd = cwd or get_cwd()
+    if not path:
+        abs_path = default or cwd
+    elif os.path.isabs(path):
+        abs_path = path
+    else:
+        abs_path = os.path.normpath(os.path.join(cwd, path))
+    return abs_path
+
+
+def _get_conanfile_path(path, cwd, py):
```

Comment:
Start incrementally, remove all old stuff, implement only the new functionality.